scripts/updateNode.ts

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env -S deno run --allow-net --allow-read --allow-write
+import { dirname, fromFileUrl, join } from "https://deno.land/std@0.98.0/path/mod.ts";
+import { compare } from "https://deno.land/x/semver@v1.4.0/mod.ts";
+
+const root = dirname(dirname(fromFileUrl(import.meta.url)));
+
+const replace = async (file: string, regexp: RegExp, replacer: string) => {
+    const path = join(root, file);
+
+    const content = await Deno.readTextFile(path);
+
+    await Deno.writeTextFile(path, content.replace(regexp, replacer));
+};
+
+interface Version {
+    version: string;
+    lts: string|false;
+}
+
+const versions: Version[] = await (await fetch('https://nodejs.org/dist/index.json')).json();
+
+const lastLts = versions
+    .filter((version) => version.lts !== false)
+    .sort((a, b) => compare(a.version, b.version))
+    .pop();
+
+if (!lastLts) {
+    throw new Error('No LTS found.');
+}
+
+const lastVersion = lastLts.version.replace(/^v/, '');
+
+// Update CircleCI config
+await replace('generators/root/templates/base/.circleci/config.yml', /image: circleci\/node:\d+\.\d+\.\d+/, `image: circleci/node:${lastVersion}`);
```

Comment:
Seems unlikely, but any chance we could not get the corresponding image from circle ? I assume latest LTS is fine but could we not fetch the data directly from circle for their available image and have a fallback ?

Reply:
Actually, it's quite likely since when a new node version is release first the official images are built, and then the CircleCI images are build so for up a to a few days the images may be not available.

But I think it's not a problem to keep the PR in a failing state for a few days and merge it when all the images are available.

Reply:
Ok I see what you mean, I feel like we should improve it by only creating the PR when the image is available for all node consumers. It would put the burden on the code rather than on the dev to check and merge the PR when appropriate. I assume this is possible by hitting some circleci endpoint or docker hub ?

Reply:
Yes we could fetch the tags from docker hub to check if the are pushed or not but:
 - it would complexify the script since we would have to fetch multiple sources and compare them to get the last version available everywhere
 - I personally prefer to have the PR opened has son as the version is release to get notified that the version is released and rely on the CI to know if the images are available.

Reply:
I don't see the script complexity as being an issue, but I see how having the PR asap can be used as a notification. Like you said a matter of personal preference at this point. This is already a great improvement.